### PR TITLE
feat(created_at_sort): add support for sorting by created

### PR DESCRIPTION
**Pull request by exp0nge**
Implements https://github.com/concourse/registry-image-resource/issues/352

### CODE REVIEW

File: check_test.go [hunk 1]
```diff
@@ -776,8 +779,28 @@ var _ = DescribeTable("tracking semver tags",
...
+			Tags: map[string]string{
+				"1.0.0":                     "random-1",
+				"non-semver-tag":            "random-2",
+				"gem-1338-git-4bd8a5e1a244": "random-3",
+				"gem-182-git-6bd8a5e1a2b3":  "random-4",
+				"gem-1337-git-4bd8a5e1a244": "random-5",
+			},
+			TagsToTime: map[string]time.Time{
+				"gem-1338-git-4bd8a5e1a244": time.Date(2024, 1, 4, 5, 0, 0,0, time.UTC),
+				"gem-181-git-6bd8a5e1a2b3":  time.Date(2024, 1, 4, 0, 0, 0,0, time.UTC),
```
Comment (xtremerui) [inline check_test.go:798]:
```suggestion
				"gem-182-git-6bd8a5e1a2b3":  time.Date(2024, 1, 4, 0, 0, 0,0, time.UTC),
```

Comment (exp0nge) [reply check_test.go]: Thanks for catching that. This should have no impact on tests as the default timestamp is set to unix epoch I believe (so 182 would always appear last).

I see the CI failed, but I can't see the output, the tests pass for my locally:
```
docker build -t registry-image-resource --target tests --build-arg base_image=docker-virtual.artifactory.m-devprod-prod.use1.eks.gem.link/paketobuildpacks/run-jammy-base:latest .

[+] Building 67.5s (24/24) FINISHED                                                                                                                                                                                             docker:desktop-linux
```

Comment (xtremerui) [reply check_test.go]: Hi @exp0nge , here is the test failure output:

```
#23 0.390 • Failure [0.036 seconds]

#23 0.391 tracking semver tags

#23 0.391 github.com/concourse/registry-image-resource/check_test.go:736

#23 0.391   regex maintain ordering [It]

#23 0.391   github.com/concourse/registry-image-resource/check_test.go:832

#23 0.391 

#23 0.391   Expected

#23 0.391       <resource.CheckResponse | len:2, cap:2>: [

#23 0.391           {

#23 0.391               Tag: "67e3c33-dev",

#23 0.391               Digest: "sha256:3f9470233e20c7aeb31ad999b7007fe09f1ea2745aeedb0410a0d17c07050524",

#23 0.391           },

#23 0.391           {

#23 0.391               Tag: "3bd8a5e-dev",

#23 0.391               Digest: "sha256:4df74f322f4b1e8ad712dbcb5ecbfc783a14d16c08973eb255c9858aff806a26",

#23 0.391           },

#23 0.391       ]

#23 0.391   to equal

#23 0.391       <resource.CheckResponse | len:2, cap:2>: [

#23 0.391           {

#23 0.391               Tag: "3bd8a5e-dev",

#23 0.391               Digest: "sha256:4df74f322f4b1e8ad712dbcb5ecbfc783a14d16c08973eb255c9858aff806a26",

#23 0.391           },

#23 0.391           {

#23 0.391               Tag: "67e3c33-dev",

#23 0.391               Digest: "sha256:3f9470233e20c7aeb31ad999b7007fe09f1ea2745aeedb0410a0d17c07050524",

#23 0.391           },

#23 0.391       ]



```

Comment (exp0nge) [reply check_test.go]: Are these tests simply flakey? I ran against "master", and it fails randomly, see:
```
❯ go test --ginkgo.focus "regex maintain ordering"
Running Suite: RegistryImageResource Suite
==========================================
Random Seed: 1710958984
Will run 1 of 117 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
Ran 1 of 117 Specs in 2.239 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 116 Skipped
PASS
ok  	github.com/concourse/registry-image-resource	2.733s
❯ go test --ginkgo.focus "regex maintain ordering"
Running Suite: RegistryImageResource Suite
==========================================
Random Seed: 1710958988
Will run 1 of 117 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• Failure [0.388 seconds]
tracking semver tags
/Users/md.islam/projects/registry-image-resource/check_test.go:733
  regex maintain ordering [It]
  /Users/md.islam/projects/registry-image-resource/check_test.go:809

  Expected
      <resource.CheckResponse | len:2, cap:2>: [
          {
              Tag: "67e3c33-dev",
              Digest: "sha256:30caa03e1092911ea0f6dc50b794456356d5dfe9c6586421897ed365849e01af",
          },
          {
              Tag: "3bd8a5e-dev",
              Digest: "sha256:873f9770e505ebd4b4a37cb3354af57c96824578899fb3485c97ef7c52ef08f4",
          },
      ]
  to equal
      <resource.CheckResponse | len:2, cap:2>: [
          {
              Tag: "3bd8a5e-dev",
              Digest: "sha256:873f9770e505ebd4b4a37cb3354af57c96824578899fb3485c97ef7c52ef08f4",
          },
          {
              Tag: "67e3c33-dev",
              Digest: "sha256:30caa03e1092911ea0f6dc50b794456356d5dfe9c6586421897ed365849e01af",
          },
      ]

  /Users/md.islam/projects/registry-image-resource/check_test.go:1270
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 1 Failure:

[Fail] tracking semver tags [It] regex maintain ordering
/Users/md.islam/projects/registry-image-resource/check_test.go:1270

Ran 1 of 117 Specs in 2.246 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 116 Skipped
--- FAIL: TestRegistryImageResource (2.27s)
FAIL
exit status 1
FAIL	github.com/concourse/registry-image-resource	2.753s
❯ git rev-parse --short HEAD
13bcb3c
```

**Comment by xtremerui** [comment]
@exp0nge we might want to fix those flaky test on our end first. This will be included in next Concourse release.

**Comment by taylorsilva** [comment]
@exp0nge I fixed the flaky check tests in #357 
You'll need to rebase on `master` and fix the merge conflicts please 🙏

**Comment by exp0nge** [comment]
> @exp0nge I fixed the flaky check tests in #357 You'll need to rebase on `master` and fix the merge conflicts please 🙏

Done!

**Comment by exp0nge** [comment]
@taylorsilva / @xtremerui What's the release schedule? I would like to stop running a custom fork 😄

**Comment by taylorsilva** [comment]
I really want this change too as it'll help with some pipelines I have.
Sadly waiting on Rui to reply here: https://github.com/concourse/concourse/issues/8938
I don't want to just barrel ahead and step on his toes. idk what his day-to-day is like right now and how much time he gets to work on Concourse.

(FYI, I'm not part of Broadcom, just an external guy like you, but with some say in the project. I work on Concourse-stuff during my own time. I have a separate day job.)

**Comment by exp0nge** [comment]
>Sadly waiting on Rui to reply here: https://github.com/concourse/concourse/issues/8938

Ah, thanks for the reference. Seems quite involved.

**Comment by risicle** [comment]
Any update on a v1.10.0?

**Pull request merged**